src/infra/core/Cargo.toml

```diff
@@ -142,6 +142,7 @@ curl-sys = { optional = true, version = "0.4" }
 datafusion-ethers = { optional = true, version = "42" }
 datafusion-functions-json = { optional = true, version = "0.42" }
 rumqttc = { optional = true, version = "0.24" }
+mockall = { optional = true, version = "0.13", default-features = false }
```

Comment:
If it's needed in dependencies, it's no longer needed in dev-dependencies.

Reply:
I have to add it twice for the following reason -- we have three use cases:
1) no `mockall` needed
2) `mockall` as a dep if `testing` feature present 
3) `mockall` as a dev dep for tests

If we remove the dev dep, `3)` won't work